application/testing/CMakeLists.txt

```diff
@@ -80,6 +80,11 @@ function(f3d_test)
     math(EXPR _timeout "3 * ${_timeout}")
   endif()
 
+  if(F3D_TEST_UI AND NOT F3D_MODULE_UI)
+    # UI tests require ImGui
+    set_tests_properties(f3d::${F3D_TEST_NAME} PROPERTIES DISABLED ON)
+  endif()
```

Comment:
I know its simpler this way but it breaks a bit the design where we do not add test that require a specific dependency, but here you disable them.

Reply:
I agree. But there are much more tests depending on imgui here (and even more in the future), so I don't think adding a test for `F3D_MODULE_UI` everywhere in the file is better

Reply:
Should we consider adding a dependency logic in the macro then ?

eg: adding `ASSIMP` or even `VTK 9.2.20221203` in the test would disable it or something ?

Definitely out of scope but that may clarify the whole logic.

Reply:
For modules (Raytracing/EXR/UI) I think I agree, for plugins I don't think so.  
Maybe plugins application tests may be even moved outside of this file to the plugin folder itself and expose a `f3dTest.cmake` for plugin developers.

Reply:
For VTK arg I agree too.

Reply:
noted, lets keep this in mind